backend/src/main/java/ai/verta/modeldb/App.java

```diff
@@ -341,4 +325,42 @@ public static void initializeTelemetryBasedOnConfig(MDBConfig mdbConfig)
       LOGGER.info("Telemetry opt out by user");
     }
   }
+
+  @PreDestroy
+  public void onShutDown() {
+    if (App.getInstance().server.isPresent()) {
+      var server = App.getInstance().server.get();
+      try {
+        int activeRequestCount = MonitoringInterceptor.ACTIVE_REQUEST_COUNT.get();
+        while (activeRequestCount > 0) {
+          activeRequestCount = MonitoringInterceptor.ACTIVE_REQUEST_COUNT.get();
+          LOGGER.info("Active Request Count in while:{} ", activeRequestCount);
+          Thread.sleep(1000); // wait for 1s
+        }
+        // Use stderr here since the logger may have been reset by its JVM shutdown
+        // hook.
+        LOGGER.info("*** Shutting down gRPC server since JVM is shutting down ***");
+        server.shutdown();
+        server.awaitTermination();
+        LOGGER.info("*** Server Shutdown ***");
+      } catch (InterruptedException e) {
+        LOGGER.error("Getting error while graceful shutdown", e);
+        // Restore interrupted state...
+        Thread.currentThread().interrupt();
+      }
+    }
+
+    initiateShutdown(0);
+
+    cleanUpPIDFile();
+  }
+
+  private void cleanUpPIDFile() {
+    var path = System.getProperty(CommonConstants.USER_DIR) + "/" + ModelDBConstants.BACKEND_PID;
+    File pidFile = new File(path);
+    if (pidFile.exists()) {
+      pidFile.deleteOnExit();
+      LOGGER.error(ModelDBConstants.BACKEND_PID + " file is deleted: {}", pidFile.exists());
```

Comment:
This will always log `true` since `deleteOnExit()` deletes the file later right?

Reply:
Yes but added log just for tracing file is deleted or not.